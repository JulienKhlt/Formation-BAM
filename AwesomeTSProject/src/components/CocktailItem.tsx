import React from 'react';
import {StyleSheet, View, Text, Image} from 'react-native';
import {cocktail} from '../../types';

interface cocktail_container {
  cocktail: cocktail;
}

export const CocktailItem = (props: cocktail_container) => {
  const cocktail: cocktail = props.cocktail;
  return (
    <View style={styles.main_container}>
      <Image style={styles.image} source={{uri: 'image'}} />
      <View style={styles.content_container}>
        <View style={styles.header_container}>
          <Text style={styles.title_text}>{cocktail.strDrink}</Text>
        </View>
        <View style={styles.description_container}>
          <Text style={styles.description_text} numberOfLines={6}>
            {cocktail.strInstructions}
          </Text>
        </View>
        <View style={styles.tag_container}>
          <Text style={styles.tag_text} numberOfLines={1}>
            {cocktail.strTags.replace(/,/g, ', ')}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  main_container: {
    height: 190,
    flexDirection: 'row',
  },
  image: {
    width: 140,
    height: 180,
    margin: 5,
    backgroundColor: 'gray',
  },
  content_container: {
    flex: 1,
    margin: 5,
  },
  header_container: {
    flex: 3,
    flexDirection: 'row',
  },
  title_text: {
    fontWeight: 'bold',
    fontSize: 20,
    flex: 1,
    flexWrap: 'wrap',
    paddingRight: 5,
  },
  description_container: {
    flex: 10,
    alignItems: 'center',
  },
  description_text: {
    fontStyle: 'italic',
    color: '#666666',
  },
  tag_container: {
    flex: 3,
  },
  tag_text: {
    textAlign: 'right',
    fontSize: 14,
  },
});
