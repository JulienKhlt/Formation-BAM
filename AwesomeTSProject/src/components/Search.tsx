import React, {useRef, useState} from 'react';
import {
  View,
  TextInput,
  Button,
  StyleSheet,
  FlatList,
  Text,
} from 'react-native';

import {cocktails} from '../../helpers/cocktailData';
import {CocktailItem} from './CocktailItem';
import {cocktail} from '../../types';
import {getCocktailsFromApiWithSearchedText} from '../../API/cocktailDB';

export const Search = () => {
  const [_cocktails, set_cocktails] = useState<cocktail[]>([]);
  const searched_text = useRef<string>('');

  const _loadCocktails = async (text: string) => {
    if (searched_text.current.length > 0) {
      const drinks = await getCocktailsFromApiWithSearchedText(text);
      console.log(drinks);

      set_cocktails(drinks);
    }
  };

  const _searchedTextInputChanged = (text: string) => {
    searched_text.current = text;
  };

  return (
    <View>
      <TextInput
        style={styles.textinput}
        accessibilityLabel="cocktail input"
        placeholder="Titre du cocktail"
        onChangeText={text => _searchedTextInputChanged(text)}
      />
      <Button
        style={styles.textinput}
        title="Rechercher"
        onPress={() => {
          _loadCocktails(searched_text.current);
        }}
      />
      <FlatList
        data={_cocktails}
        keyExtractor={item => item.idDrink.toString()}
        renderItem={({item}) => <CocktailItem cocktail={item} />}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  textinput: {
    marginLeft: 5,
    marginRight: 5,
    height: 50,
    borderColor: '#000000',
    borderWidth: 1,
    paddingLeft: 5,
  },
});
