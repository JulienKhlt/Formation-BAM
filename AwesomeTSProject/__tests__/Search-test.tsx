// __tests__/Search-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import {Search} from '../src/components/Search';

test('search renders correctly', () => {
  const search = renderer.create(<Search />).toJSON();
  expect(search).toMatchSnapshot();
});
