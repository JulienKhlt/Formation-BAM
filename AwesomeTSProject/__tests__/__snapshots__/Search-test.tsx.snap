// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search renders correctly 1`] = `
<View>
  <TextInput
    accessibilityLabel="cocktail input"
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Titre du cocktail"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#000000",
        "borderWidth": 1,
        "height": 50,
        "marginLeft": 5,
        "marginRight": 5,
        "paddingLeft": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Rechercher
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "dateModified": "2016-11-04 09:17:09",
          "idDrink": "11000",
          "strAlcoholic": "Alcoholic",
          "strCategory": "Cocktail",
          "strCreativeCommonsConfirmed": "Yes",
          "strDrink": "Mojito",
          "strDrinkAlternate": null,
          "strDrinkThumb": "https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg",
          "strGlass": "Highball glass",
          "strIBA": "Contemporary Classics",
          "strImageAttribution": "anilaha https://pixabay.com/users/anilaha-16242978/",
          "strImageSource": "https://pixabay.com/photos/cocktail-mojito-cocktail-recipe-5096281/",
          "strIngredient1": "Light rum",
          "strIngredient10": null,
          "strIngredient11": null,
          "strIngredient12": null,
          "strIngredient13": null,
          "strIngredient14": null,
          "strIngredient15": null,
          "strIngredient2": "Lime",
          "strIngredient3": "Sugar",
          "strIngredient4": "Mint",
          "strIngredient5": "Soda water",
          "strIngredient6": null,
          "strIngredient7": null,
          "strIngredient8": null,
          "strIngredient9": null,
          "strInstructions": "Muddle mint leaves with sugar and lime juice. Add a splash of soda water and fill the glass with cracked ice. Pour the rum and top with soda water. Garnish and serve with straw.",
          "strInstructionsDE": "Minzblätter mit Zucker und Limettensaft verrühren. Füge einen Spritzer Sodawasser hinzu und fülle das Glas mit gebrochenem Eis. Den Rum eingießen und mit Sodawasser übergießen. Garnieren und mit einem Strohhalm servieren.",
          "strInstructionsES": null,
          "strInstructionsFR": null,
          "strInstructionsIT": "Pestare le foglie di menta con lo zucchero e il succo di lime.
Aggiungere una spruzzata di acqua di seltz e riempi il bicchiere con ghiaccio tritato.
Versare il rum e riempire con acqua di seltz.
Guarnire con una fetta di lime, servire con una cannuccia.",
          "strInstructionsZH-HANS": null,
          "strInstructionsZH-HANT": null,
          "strMeasure1": "2-3 oz ",
          "strMeasure10": null,
          "strMeasure11": null,
          "strMeasure12": null,
          "strMeasure13": null,
          "strMeasure14": null,
          "strMeasure15": null,
          "strMeasure2": "Juice of 1 ",
          "strMeasure3": "2 tsp ",
          "strMeasure4": "2-4 ",
          "strMeasure5": null,
          "strMeasure6": null,
          "strMeasure7": null,
          "strMeasure8": null,
          "strMeasure9": null,
          "strTags": "IBA,ContemporaryClassic,Alcoholic,USA,Asia,Vegan,Citrus,Brunch,Hangover,Mild",
          "strVideo": null,
        },
        Object {
          "dateModified": "2016-11-04 09:24:43",
          "idDrink": "15841",
          "strAlcoholic": "Alcoholic",
          "strCategory": "Cocktail",
          "strCreativeCommonsConfirmed": "No",
          "strDrink": "Mojito Extra",
          "strDrinkAlternate": null,
          "strDrinkThumb": "https://www.thecocktaildb.com/images/media/drink/vwxrsw1478251483.jpg",
          "strGlass": "Collins glass",
          "strIBA": null,
          "strImageAttribution": null,
          "strImageSource": null,
          "strIngredient1": "Mint",
          "strIngredient10": null,
          "strIngredient11": null,
          "strIngredient12": null,
          "strIngredient13": null,
          "strIngredient14": null,
          "strIngredient15": null,
          "strIngredient2": "Lemon juice",
          "strIngredient3": "Dark rum",
          "strIngredient4": "Club soda",
          "strIngredient5": "Angostura bitters",
          "strIngredient6": null,
          "strIngredient7": null,
          "strIngredient8": null,
          "strIngredient9": null,
          "strInstructions": "Put mint with lemon juice in a glas, mash the mint with a spoon, ice, rum & fill up with club soda. Top it with Angostura.",
          "strInstructionsDE": "Minze mit Zitronensaft in ein Glas geben, Minze mit einem Löffel zerdrücken. Eis und Rum hinzufügen und mit Soda auffüllen. Krönen Sie es mit Angostura.",
          "strInstructionsES": null,
          "strInstructionsFR": null,
          "strInstructionsIT": "Metti la menta con il succo di limone in un bicchiere, schiaccia la menta con un cucchiaio, ghiaccio, rum e riempila con la soda.
Completalo con Angostura.",
          "strInstructionsZH-HANS": null,
          "strInstructionsZH-HANT": null,
          "strMeasure1": "1/2 handful ",
          "strMeasure10": null,
          "strMeasure11": null,
          "strMeasure12": null,
          "strMeasure13": null,
          "strMeasure14": null,
          "strMeasure15": null,
          "strMeasure2": "3 cl ",
          "strMeasure3": "1/8 L Jamaican ",
          "strMeasure4": "1/8 L ",
          "strMeasure5": "8 drops ",
          "strMeasure6": null,
          "strMeasure7": null,
          "strMeasure8": null,
          "strMeasure9": null,
          "strTags": null,
          "strVideo": null,
        },
        Object {
          "dateModified": null,
          "idDrink": "178336",
          "strAlcoholic": "Alcoholic",
          "strCategory": "Cocktail",
          "strCreativeCommonsConfirmed": "No",
          "strDrink": "Blueberry Mojito",
          "strDrinkAlternate": null,
          "strDrinkThumb": "https://www.thecocktaildb.com/images/media/drink/07iep51598719977.jpg",
          "strGlass": "Highball glass",
          "strIBA": null,
          "strImageAttribution": null,
          "strImageSource": null,
          "strIngredient1": "Dark Rum",
          "strIngredient10": null,
          "strIngredient11": null,
          "strIngredient12": null,
          "strIngredient13": null,
          "strIngredient14": null,
          "strIngredient15": null,
          "strIngredient2": "Lime Juice",
          "strIngredient3": "Sugar",
          "strIngredient4": "Blueberries",
          "strIngredient5": "Lemon-lime soda",
          "strIngredient6": "",
          "strIngredient7": "",
          "strIngredient8": null,
          "strIngredient9": null,
          "strInstructions": "Muddle the blueberries with the other ingredients and serve in a highball glass. Garnish with mint and a half slice of lime.",
          "strInstructionsDE": null,
          "strInstructionsES": null,
          "strInstructionsFR": null,
          "strInstructionsIT": "Pestare i mirtilli con gli altri ingredienti e servire in un bicchiere highball. Guarnire con la menta e mezza fetta di lime.",
          "strInstructionsZH-HANS": null,
          "strInstructionsZH-HANT": null,
          "strMeasure1": "2 shots",
          "strMeasure10": null,
          "strMeasure11": null,
          "strMeasure12": null,
          "strMeasure13": null,
          "strMeasure14": null,
          "strMeasure15": null,
          "strMeasure2": "1 shot",
          "strMeasure3": "Dash",
          "strMeasure4": "Whole",
          "strMeasure5": "Top",
          "strMeasure6": "",
          "strMeasure7": "",
          "strMeasure8": null,
          "strMeasure9": null,
          "strTags": null,
          "strVideo": null,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 190,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "image",
              }
            }
            style={
              Object {
                "backgroundColor": "gray",
                "height": 180,
                "margin": 5,
                "width": 140,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "flexWrap": "wrap",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "paddingRight": 5,
                  }
                }
              >
                Mojito
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 10,
                }
              }
            >
              <Text
                numberOfLines={6}
                style={
                  Object {
                    "color": "#666666",
                    "fontStyle": "italic",
                  }
                }
              >
                Muddle mint leaves with sugar and lime juice. Add a splash of soda water and fill the glass with cracked ice. Pour the rum and top with soda water. Garnish and serve with straw.
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 3,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "fontSize": 14,
                    "textAlign": "right",
                  }
                }
              >
                IBA, ContemporaryClassic, Alcoholic, USA, Asia, Vegan, Citrus, Brunch, Hangover, Mild
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 190,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "image",
              }
            }
            style={
              Object {
                "backgroundColor": "gray",
                "height": 180,
                "margin": 5,
                "width": 140,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "flexWrap": "wrap",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "paddingRight": 5,
                  }
                }
              >
                Mojito Extra
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 10,
                }
              }
            >
              <Text
                numberOfLines={6}
                style={
                  Object {
                    "color": "#666666",
                    "fontStyle": "italic",
                  }
                }
              >
                Put mint with lemon juice in a glas, mash the mint with a spoon, ice, rum & fill up with club soda. Top it with Angostura.
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 190,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "image",
              }
            }
            style={
              Object {
                "backgroundColor": "gray",
                "height": 180,
                "margin": 5,
                "width": 140,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "flexWrap": "wrap",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "paddingRight": 5,
                  }
                }
              >
                Blueberry Mojito
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 10,
                }
              }
            >
              <Text
                numberOfLines={6}
                style={
                  Object {
                    "color": "#666666",
                    "fontStyle": "italic",
                  }
                }
              >
                Muddle the blueberries with the other ingredients and serve in a highball glass. Garnish with mint and a half slice of lime.
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
